src/pcre2_jit_compile.c

```diff
@@ -10898,7 +10898,8 @@ if (opcode == OP_ASSERT || opcode == OP_ASSERTBACK)
     {
     JUMPTO(SLJIT_JUMP, backtrack->matchingpath);
     JUMPHERE(brajump);
-    if (framesize >= 0)
+    SLJIT_ASSERT(framesize != 0);
+    if (framesize > 1)
```

Comment:
framesize == 1 should be valid

Reply:
but the code it generates is useless, no?

Reply:
Why? Not just `>= 2` cases are valid, the `== 1` is valid as well. `>= 0` can be turned to `> 0` or `>= 1`

Reply:
Is doing something like this worth it then?

```
diff --git a/src/pcre2_jit_compile.c b/src/pcre2_jit_compile.c
index ca13cd76..dc78f52a 100644
--- a/src/pcre2_jit_compile.c
+++ b/src/pcre2_jit_compile.c
@@ -10904,7 +10904,8 @@ if (opcode == OP_ASSERT || opcode == OP_ASSERTBACK)
       OP1(SLJIT_MOV, STACK_TOP, 0, SLJIT_MEM1(SLJIT_SP), private_data_ptr);
       add_jump(compiler, &common->revertframes, JUMP(SLJIT_FAST_CALL));
       OP1(SLJIT_MOV, TMP1, 0, SLJIT_MEM1(STACK_TOP), STACK(-2));
-      OP2(SLJIT_ADD, STACK_TOP, 0, STACK_TOP, 0, SLJIT_IMM, (framesize - 1) * sizeof(sljit_sw));
+      if (framesize > 1)
+        OP2(SLJIT_ADD, STACK_TOP, 0, STACK_TOP, 0, SLJIT_IMM, (framesize - 1) * sizeof(sljit_sw));
       OP1(SLJIT_MOV, SLJIT_MEM1(SLJIT_SP), private_data_ptr, TMP1, 0);
       }
     set_jumps(backtrack->common.own_backtracks, LABEL());
```